### PR TITLE
feat(broot): add broot spec

### PR DESCRIPTION
**Pull request by Pooja444**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
New spec for `broot`

**What is the current behavior? (You can also link to an open issue here)**
Missing spec for `broot` (closes https://github.com/withfig/autocomplete/issues/1526)

**What is the new behavior (if this is a feature change)?**
Add spec for `broot`

**Additional info:**
- Ran `npm run test` and `npm run lint:fix` before raising PR
- Tested this spec locally, autocomplete is working correctly

### CODE REVIEW

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
@Pooja444 thank you for the PR! Beautifully done.

This is embarrassing: I ran `broot --help` instead of looking at the man-page and... there's actually a bunch more options that I didn't know about 🤦 Would you be able to add those? If you're short on time, I'm happy to land this PR as-is and we can add the rest later 🙂

```
broot 1.16.0
dystroy <denys.seguret@gmail.com>
A tree explorer and a customizable launcher

Complete documentation lives at https://dystroy.org/broot"

USAGE:
    broot [OPTIONS] [FILE]

ARGS:
    <FILE>
            Root Directory

OPTIONS:
    -c, --cmd <CMD>
            Semicolon separated commands to execute

        --color <COLOR>
            Whether to have styles and colors (auto is default and usually OK)
            
            [default: auto]
            [possible values: auto, yes, no]

        --conf <CONF>
            Semicolon separated paths to specific config files"),

    -d, --dates
            Show the last modified date of files and directories"

    -D, --no-dates
            Don't show the last modified date"

    -f, --only-folders
            Only show folders

    -F, --no-only-folders
            Show folders and files alike

    -g, --show-git-info
            Show git statuses on files and stats on repo

    -G, --no-show-git-info
            Don't show git statuses on files and stats on repo

        --get-root
            Ask for the current root of the remote broot

        --git-status
            Only show files having an interesting git status, including hidden ones

    -h, --hidden
            Show hidden files

    -H, --no-hidden
            Don't show hidden files

        --height <HEIGHT>
            Height (if you don't want to fill the screen or for file export)

        --help
            Print help information

    -i, --git-ignored
            Show git ignored files

    -I, --no-git-ignored
            Don't show git ignored files

        --install
            Install or reinstall the br shell function

        --listen <LISTEN>
            A socket to listen to for commands

        --no-sort
            Don't sort

        --outcmd <OUTCMD>
            Where to write the produced cmd (if any)

    -p, --permissions
            Show permissions

    -P, --no-permissions
            Don't show permissions

        --print-shell-function <PRINT_SHELL_FUNCTION>
            Print to stdout the br function for a given shell

    -s, --sizes
            Show the size of files and directories

    -S, --no-sizes
            Don't show sizes

        --send <SEND>
            A socket that broot sends commands to before quitting

        --set-install-state <SET_INSTALL_STATE>
            Where to write the produced cmd (if any)
            
            [possible values: undefined, refused, installed]

        --show-root-fs
            Show filesystem info on top

        --sort-by-count
            Sort by count (only show one level of the tree)

        --sort-by-date
            Sort by date (only show one level of the tree)

        --sort-by-size
            Sort by size (only show one level of the tree)

        --sort-by-type
            Same as sort-by-type-dirs-first

        --sort-by-type-dirs-first
            Sort by type, directories first (only show one level of the tree)

        --sort-by-type-dirs-last
            Sort by type, directories last (only show one level of the tree)

    -t, --trim-root
            Trim the root too and don't show a scrollbar

    -T, --no-trim-root
            Don't trim the root level, show a scrollbar

    -V, --version
            Print version information

    -w, --whale-spotting
            Sort by size, show ignored and hidden files

        --write-default-conf <WRITE_DEFAULT_CONF>
            Write default conf files in given directory
```

**Comment by SeparateRecords** [comment]
Sounds great to me! No rush at all. Thank you again for working on this!

**Comment by SeparateRecords** [comment]
(and sorry for not checking this beforehand!)

**Comment by Pooja444** [comment]
@SeparateRecords I have added the rest of the options and args, please let me know if it looks good, thanks a lot!

File: src/broot.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    "Broot lets you explore file hierarchies with a tree-like view, manipulate files, launch actions, and define your own shortcuts. broot is best launched as br: this shell function gives you access to more commands, especially cd. The br shell function is interactively installed on first broot launch. Flags and options can be classically passed on launch but also written in the configuration file. Each flag has a counter-flag so that you can cancel at command line a flag which has been set in the configuration file",
+  options: [
+    {
+      name: ["--dates", "-d"],
+      description: "Show the last modified date of files and directories",
+    },
+    {
+      name: ["--no-dates", "-D"],
+      description: "Don't show the last modified date",
+    },
```
Comment (fedeci) [inline src/broot.ts:13]: These are mutually [`exclusiveOn`](https://fig.io/docs/reference/option#exclusiveon). Same for all the other options that has a `--` and a `--no-` version. Can you add that property?

File: src/broot.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        "Height to use if you don't want to fill the screen or for file export (by default the terminal height is used)",
+      args: {
+        name: "height",
+      },
+    },
+    {
+      name: ["--out", "-o"],
+      description: "Where to write the produced path, if any",
+      args: {
+        name: "file-export-path",
```
Comment (fedeci) [inline src/broot.ts:161]:
```suggestion
        name: "file-export-path",
        template: "filepaths",
        suggestCurrentToken: true,
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks for this PR, it looks great! Can you address the comments please?

**Comment by Pooja444** [comment]
Thanks for your suggestions @fedeci! I have updated my pull request as per your comments, please review and let me know if it looks good

**Review by SeparateRecords — APPROVED** [review]
LGTM! Thank you @Pooja444 🙂

**Pull request merged**